internal/bufferedwrites/upload_handler.go

```diff
@@ -123,6 +138,7 @@ func (uh *UploadHandler) Finalize() error {
 	err := uh.writer.Close()
 	if err != nil {
 		logger.Errorf("UploadHandler.Finalize(): %v", err)
+		close(uh.signalUploadFailure)
```

Comment:
why do you need this in finalize? can't finalize return error directly?

Reply:
We are returning the error as well (next line). Added this to ensure that any follow up writes on the same inode fail after this error.

Reply:
In what scenario do we get follow up writes. As soon as the finalize is triggered (success or failure) - bwh should be set to nil.

Reply:
Removed `close(uh.signalUploadFailure)` as discussed offline that we will not solve for multithreaded writes for now.
I will take care of making bwh nil in the follow up PR where I integrate flush with file inode sync changes.